crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -1493,127 +1533,183 @@ end
+    # convert the index into a memory offset
+    mul.ACCOUNT_STORAGE_SLOT_DATA_LENGTH
+    # => [offset]
+
     # get account storage slots section offset
-    exec.memory::get_account_storage_slots_section_ptr
-    # => [acct_storage_slots_section_offset, index]
+    exec.memory::get_native_account_storage_slots_ptr
+    # => [storage_slots_ptr, offset]
 
-    exec.get_item_by_index_raw
-    # => [VALUE]
+    dup.1 add dup
+    # => [slot_ptr, slot_ptr, offset]
+
+    # load the name ID
+    add.ACCOUNT_SLOT_NAME_ID_SUFFIX_OFFSET mem_load
+    # => [name_id_suffix, slot_ptr, offset]
+
+    dup.1 add.ACCOUNT_SLOT_NAME_ID_PREFIX_OFFSET mem_load
+    # => [name_id_prefix, name_id_suffix, slot_ptr, offset]
+
+    # load the current value
+    movup.2 exec.get_item_raw
+    # => [CURRENT_VALUE, name_id_prefix, name_id_suffix, offset]
+
+    # get account initial storage slots section offset
+    exec.memory::get_account_initial_storage_slots_ptr
+    # => [init_storage_slots_ptr, CURRENT_VALUE, name_id_prefix, name_id_suffix, offset]
+
+    movup.7 add
+    # => [init_slot_ptr, CURRENT_VALUE, name_id_prefix, name_id_suffix]
+
+    exec.get_item_raw
+    # => [INITIAL_VALUE, CURRENT_VALUE, name_id_prefix, name_id_suffix]
 end
 
-#! Gets the initial item from the storage slot at the provided index.
+#! Gets the name of the storage slot at the provided index.
 #!
 #! WARNING: The index must be in bounds.
 #!
 #! Inputs:  [index]
-#! Outputs: [VALUE]
+#! Outputs: [name_id_prefix, name_id_suffix]
 #!
 #! Where:
-#! - storage_slots_ptr is the pointer to the storage slots section.
-#! - index is the index of the item to get.
-#! - VALUE is the value of the item.
-export.get_initial_item_by_index
-    # get account initial storage slots section offset
-    exec.memory::get_account_initial_storage_slots_ptr
-    # => [account_initial_storage_slots_ptr, index]
+#! - index is the index of the slot.
+#! - name_id_{prefix, suffix} are the prefix and suffix felts of the name identifier, which are
+#!   the first two felts of the hashed slot name.
+export.get_slot_name
+    # convert the index into a memory offset
+    mul.ACCOUNT_STORAGE_SLOT_DATA_LENGTH
+    # => [offset]
 
-    exec.get_item_by_index_raw
-    # => [VALUE]
+    exec.memory::get_account_storage_slots_section_ptr
+    add
+    # => [slot_ptr]
+
+    dup add.ACCOUNT_SLOT_NAME_ID_SUFFIX_OFFSET mem_load
+    # => [name_id_suffix, slot_ptr]
+
+    swap add.ACCOUNT_SLOT_NAME_ID_PREFIX_OFFSET mem_load
+    # => [name_id_prefix, name_id_suffix]
 end
 
-#! Gets an item from storage using the provided storage slots pointer and index.
+#! Sets the value of the provided slot pointer to the provided value.
 #!
-#! WARNING: The index must be in bounds.
+#! The provided pointer must be a valid pointer into the current storage slot section.
+#!
+#! Inputs:  [slot_ptr, VALUE]
+#! Outputs: []
 #!
-#! Inputs:  [storage_slots_ptr, index]
+#! Where:
+#! - slot_ptr is the pointer to the storage slots section.
```

Comment:
The description here is a bit confusing: `slot_ptr` is a pointer to the storage slot, not the storage slots section.